tests/unit/models/keras/test_sampler.py

```diff
@@ -100,3 +103,77 @@ def test_ensemble_trajectory_sampler_resample_provides_new_samples_without_retra
     npt.assert_array_less(1e-4, tf.abs(evals_1 - evals_2))
     npt.assert_array_less(1e-4, tf.abs(evals_2 - evals_3))
     npt.assert_array_less(1e-4, tf.abs(evals_1 - evals_3))
+
+
+@pytest.mark.parametrize("num_evals", [10, 20])
+def test_dropout_trajectory_sampler_returns_trajectory_function_with_correctly_shaped_output(
+    num_evals: int,
+) -> None:
+    example_data = empty_dataset([1], [1])
+    test_data_1 = tf.linspace([-10.0], [10.0], num_evals)
+    test_data_2 = tf.linspace([10.0], [-10.0], num_evals)
+    test_data = tf.stack([test_data_1, test_data_2], axis=-2) # [N, B, d]
+
+    model, _, _ = trieste_mcdropout_model(example_data, dropout=DropoutNetwork)
+
+    sampler = DropoutTrajectorySampler(model)
+    trajectory = sampler.get_trajectory()
+
+    assert trajectory(test_data).shape == (num_evals, 2)
+
+
+def test_dropout_trajectory_sampler_returns_deterministic_trajectory() -> None:
```

Comment:
Leaving a note here to remind about testing the conintuous thompson sampler in the bayesian optimization integration test.